src/schema/comments.ts

```diff
@@ -665,6 +670,30 @@ export const resolvers: IResolvers<any, Context> = {
 
       return markdown.render(trimmed, { mentions });
     },
+    comment: async (
+      _,
+      { id }: { id: string },
+      ctx,
+      info,
+    ): Promise<GQLComment> => {
+      const comment = await ctx.con.getRepository(Comment).findOneOrFail({
```

Comment:
Optimization suggestion: instead of querying the comment, query it via graphorm like you do at the end. Keep the result and use the returned id to fetch the post and ensure permissions and then just return this object. Will save one call to the database

Reply:
Awesome. Will do that. I just thought that the GraphORM might not work for the expected result if not called from return. Let me update it quickly.